### PR TITLE
Ledger: improve hw wallet splash screen

### PR DESCRIPTION
**Pull request by selsta**
Screenshot 2019-08-16 at 21 12 49

### CODE REVIEW

**Comment by Realchacal** [comment]
Nice!!!!!!!!

**Comment by selsta** [comment]
Updated PR, it now also shows while transacting. Only wallet opening is missing now.

**Comment by Realchacal** [comment]
Nice!!! And hard to understand why the opening screen is any more difficult to change? (I'm not complaining in any way, I really just wonder)

**Comment by selsta** [comment]
To find out if a wallet is a Ledger wallet is has to get opened first, the splash screen gets displayed before a wallet is opened.

**Comment by Realchacal** [comment]
Oh, I see! Interesting, thanks!

**Comment by selsta** [comment]
> This could be solved if we went with the alternative proposal shown in #2324.

This wouldn’t work as I need to differentiate between Trezor and Ledger. Trezor already shows a splash screen if user input is required. So I’d need to know if it’s a Ledger wallet specifically.

**Comment by Realchacal** [comment]
I see, thanks! (By the way that's why I suggested "...your input **may** be required..." - so that it will work with different hardware wallets in the future and also if a hardware wallet has an option like "export public view key without confirmation")

File: main.qml [hunk 1]
```diff
@@ -859,7 +859,9 @@ ApplicationWindow {
                     ", priority: ", priority,
                     ", description: ", description);
 
-        showProcessingSplash("Creating transaction");
+        var splashMsg = qsTr("Creating transaction...");
+        splashMsg += appWindow.currentWallet.isLedger() ? qsTr("\n\nPlease check your hardware wallet –\nyour input may be required.") : "";
```
Comment (xiphon) [inline main.qml]: Do we need to end the sentence with a period here?

Comment (Realchacal) [reply main.qml]: I'd be fine with or without period.

**Comment by Realchacal** [comment]
I wrote the new sentence under the assumption that it must fit to all cases (Ledger/ Trezor/ other future hardware wallets, which have an option like: `[x] export public view key without confirmation.` ) If that is **not** the case (as it does look now), the sentence isn't that good anymore. Better:

- Please confirm the export of your view key on your hardware wallet.
(If you deny it, computations will have to be done on your hardware wallet, which may take an extreme long time.)

And if we wanted it perfect, we don't need "Creating wallet from device..." at this point IMO.

**Comment by Realchacal** [comment]
I phrased the added sentence under the assumption that it must fit in all cases, even if no user action is required. But if my assumption is wrong (as it seems), we'd better go like this:

- **Please confirm the export of your view key on your hardware wallet.**
(If you deny it, computations will have to be done on your hardware wallet, which may take a very long time.)

Not sure if we should keep the part in brackets. Because denying export view key is clearly a feature for pro users who know what they are doing since this may easily take weeks (see #2344).

For the transaction splash screen:
- **Please confirm the transaction details on your hardware wallet.**
(The change amount that is shown to you will not be spendable for ~20 minutes.)

Not sure if we should keep the part in brackets. But since the change amount may be a huge number this confirmation could easily cause anxiety for an average user who does not understand what this means.

File: main.qml (hunk 1 above)
Comment (selsta) [reply main.qml]: I think it looks better with a period, but I don’t care really.

**Comment by selsta** [comment]
@Realchacal I prefer it being like it is in this PR.

> Please confirm the transaction details on your hardware wallet.

This can confuse the user. It takes up to 10 seconds until the dialog shows up on the Ledger.

**Comment by Realchacal** [comment]
> It takes up to 10 seconds until the dialog shows up on the Ledger.

Strong argument, I wasn't aware! Thanks.

**Pull request merged**